.github/workflows/resource-cleanup.yml

```diff
@@ -53,3 +53,76 @@ jobs:
         run: |
           python -m pip install -r requirements.txt
           python cleaner.py
+
+  cleanup-k8s-on-ec2:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        instance: [
+          { repo_name: 'amazon-cloudwatch-agent-operator', ec2_name: 'cw-agent-operator-release', language: 'java' },
+          { repo_name: 'amazon-cloudwatch-agent-operator', ec2_name: 'cw-agent-operator-release', language: 'python' },
+          { repo_name: 'amazon-cloudwatch-agent', ec2_name: 'cw-agent-release', language: 'java' },
+          { repo_name: 'amazon-cloudwatch-agent', ec2_name: 'cw-agent-release', language: 'python' },
+          { repo_name: 'aws-otel-python-instrumentation', ec2_name: 'adot-python-release', language: 'python' },
+          { repo_name: 'aws-otel-java-instrumentation', ec2_name: 'adot-java-release', language: 'java' },
+          { repo_name: 'aws-application-signals-test-framework', ec2_name: 'python-canary', language: 'python' },
+          { repo_name: 'aws-application-signals-test-framework', ec2_name: 'java-canary', language: 'java' } ]
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check if today is the first day of the month
+        id: check_day
+        run: |
+          echo "is_first_day=$(( $(date +%d) == 01 ))" >> $GITHUB_OUTPUT
+
+      - name: Configure AWS Credentials
+        if: steps.check_day.outputs.is_first_day == '1'
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.APPLICATION_SIGNALS_E2E_TEST_ACCOUNT_ID }}:role/${{ secrets.APPLICATION_SIGNALS_E2E_TEST_ROLE_NAME }}
+          aws-region: us-east-1
+
+      - name: Delete Key Pair and EC2 Instance
+        if: steps.check_day.outputs.is_first_day == '1'
+        run: |
+          STORAGE_SECRET_KEYS=(
+            "e2e-test/${{ matrix.instance.repo_name }}/${{ matrix.instance.language }}-k8s-master-node-endpoint-temporary-storage"
+          )
+          endpoint_secret=$(aws secretsmanager get-secret-value --secret-id "${STORAGE_SECRET_KEYS[0]}" --query SecretString --output text)
+          
+          instance_name=$(aws ec2 describe-instances \
+            --filters "Name=ip-address,Values=$endpoint_secret" \
+            --query "Reservations[*].Instances[*].{Name:Tags[?Key=='Name']|[0].Value}" \
+            --output text)
```

Comment:
So we check the temp secrets for old clusters, then delete those?

We should be checking all instances associated with k8s clusters and cross referencing them with the clusters we keep for releases and canaries. If they're not in that list, then delete them.